lib/ssh_scan/scan_engine.rb

```diff
@@ -133,34 +135,34 @@ def scan(opts)
       workers.map(&:join)
 
       # Add all the fingerprints to our peristent FingerprintDatabase
-      fingerprint_db = SSHScan::FingerprintDatabase.new(opts[:fingerprint_database])
-      results.each do |result|
-        fingerprint_db.clear_fingerprints(result[:ip])
-        if result['fingerprints']
-          result['fingerprints'].values.each do |host_key_algo|
-            host_key_algo.values.each do |fingerprint|
-              fingerprint_db.add_fingerprint(fingerprint, result[:ip])
-            end
-          end
-        end
-      end
+      # fingerprint_db = SSHScan::FingerprintDatabase.new(opts[:fingerprint_database])
+      # results.each do |result|
+      #   fingerprint_db.clear_fingerprints(result[:ip])
+      #   if result['fingerprints']
+      #     result['fingerprints'].values.each do |host_key_algo|
+      #       host_key_algo.values.each do |fingerprint|
+      #         fingerprint_db.add_fingerprint(fingerprint, result[:ip])
+      #       end
+      #     end
+      #   end
+      # end
 
       # Decorate all the results with duplicate keys
-      results.each do |result|
-        if result['fingerprints']
-          ip = result[:ip]
-          result['duplicate_host_key_ips'] = []
-          result['fingerprints'].values.each do |host_key_algo|
-            host_key_algo.values.each do |fingerprint|
-              fingerprint_db.find_fingerprints(fingerprint).each do |other_ip|
-                next if ip == other_ip
-                result['duplicate_host_key_ips'] << other_ip
-              end
-            end
-          end
-          result['duplicate_host_key_ips'].uniq!        
-        end
-      end
+      # results.each do |result|
+      #   if result['fingerprints']
+      #     ip = result[:ip]
+      #     result['duplicate_host_key_ips'] = []
+      #     result['fingerprints'].values.each do |host_key_algo|
+      #       host_key_algo.values.each do |fingerprint|
+      #         fingerprint_db.find_fingerprints(fingerprint).each do |other_ip|
+      #           next if ip == other_ip
+      #           result['duplicate_host_key_ips'] << other_ip
+      #         end
+      #       end
+      #     end
+      #     result['duplicate_host_key_ips'].uniq!
+      #   end
+      # end
```

Comment:
Removed fingerprinting from here temporarily, we'll need to further abstract fingerprint and compliance decorators so they are decorated by the API itself and not workers.